### PR TITLE
Support both case sensitive, and case-insensitive search in non-fast-…

### PR DESCRIPTION
**Pull request by baldersheim**
…search attributes.

@toregge or @geirst PR

### CODE REVIEW

**Review by toregge — APPROVED** [review]

File: searchlib/src/vespa/searchlib/attribute/stringbase.h [hunk 1]
```diff
@@ -16,6 +16,27 @@
 
 namespace search {
 
+class StringSearchHelper {
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/attribute/stringbase.h:19]: Please add class comment.

File: searchlib/src/tests/attribute/stringattribute/stringattribute_test.cpp [hunk 2]
```diff
@@ -408,4 +409,83 @@ TEST("testSingleValue")
...
+TEST("test uncased match") {
+    QueryTermUCS4 xyz("xyz", QueryTermSimple::Type::WORD);
+    StringSearchHelper helper(xyz, false);
+    EXPECT_FALSE(helper.isCased());
+    EXPECT_FALSE(helper.isPrefix());
+    EXPECT_FALSE(helper.isRegex());
+    EXPECT_FALSE(helper.isMatch("axyz"));
+    EXPECT_FALSE(helper.isMatch("xyza"));
+    EXPECT_TRUE(helper.isMatch("xyz"));
+    EXPECT_TRUE(helper.isMatch("XyZ"));
```
Comment (geirst) [inline searchlib/src/tests/attribute/stringattribute/stringattribute_test.cpp:421]: Consider adding a test case where the input string is shorter than the query term. Applies for all tests.

File: searchlib/src/vespa/searchlib/attribute/stringbase.h (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/attribute/stringbase.h:19]: Fixed

File: searchlib/src/tests/attribute/stringattribute/stringattribute_test.cpp (hunk 2 above)
Comment (baldersheim) [reply searchlib/src/tests/attribute/stringattribute/stringattribute_test.cpp:421]: Fixed

**Pull request merged**